packages/react-components/src/DocsComponents/Toc.stories.tsx

```diff
@@ -37,26 +69,57 @@ const useTocStyles = makeStyles({
 // };
 
 const navigate = (url: string) => {
-  console.log('Navigate', url);
   addons.getChannel().emit(NAVIGATE_URL, url);
 };
 
 export const nameToHash = (id: string): string => id.toLowerCase().replace(/[^a-z0-9]/gi, '-');
 
-export const Toc = ({ stories }: { stories: PublishedStoreItem[] }) => {
+export const Toc = ({ stories }: { stories: TocItem[] }) => {
+  const [selected, setSelected] = React.useState('');
+
+  const handleObserver = (entries: IntersectionObserverEntry[]) => {
```

Comment:
nit: I'd just use the handler inline in the observer constructor, since you're never reusing it again, and it's being recreated on each render